mdtraj/core/topology.py

```diff
@@ -1715,76 +1848,76 @@ def __eq__(self, other):
             return False
         return True
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         """A quick comparison."""
         return self.index
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"{self.residue}-{self.name}"
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return str(self)
 
 
 # Enumerated values for bond type
 
 
 class Single(Singleton):
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "Single"
 
-    def __float__(self):
+    def __float__(self) -> float:
         return 1.0
 
 
-Single = Single()
+Single: Singleton = Single()
```

Comment:
The class is defined then immediately written over with an instance of the class (and this is why `mypy` is giving errors about `Single` and `Double`, etc being defined twice). It's kind of a weird pattern.

The easiest option is to tell `mypy` to ignore this error using `#type: ignore[no-redef]`

Another option would be to restructure the code. However, for the sake of this PR, I'd say the tag is the most appropriate.